pkg/metrics/metricbuilder/metricbuilder.go

```diff
@@ -0,0 +1,64 @@
+package metricbuilder
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// MetricBuilder is the basic metric type that must be used by any potential metric added to
+// the system. Any metric added must have the following functions which would be used in the
+// workflow.
+type MetricBuilder struct {
```

Comment:
Has no useable public API. What is this package providing?

Reply:
build function should be public. Fixing it.

Reply:
> Has no useable public API. What is this package providing?
 
Good point. I missed this on my first review.  The idea is that a higher-level metrics package will initialize instances of this type and provide an API to (e.g) pkg/asset and cmd for adding label key/value pairs and metrics values.

To achieve that I think we need to either 
* export fields from the MetricBuilder type. 
* provide setter/functions
* a combination of previous two ideas (e.g. export `values` and `labelKeyValues` and provide an initialization function)
* combine this with main metrics package

Reply:
The idea before was to put them all up into one package and use them as is but I thought of splitting them up now. I think providing setter methods will do the trick.

Reply:
>  I think providing setter methods will do the trick.

Most of the fields should be set upon initialization. creating individual setters would make initialization clunky. I don't know if setters is the right path but if it is, I would create a function for initializing, one for modifying `value` and one for modifying `labelKeyValues`